packages/backend/src/models/ProjectModel/ProjectModel.ts

```diff
@@ -1659,9 +1659,11 @@ export class ProjectModel {
                     )
                     .returning('*');
             }
-            const dashboardTabsMapping = dashboardTabs.map((c, i) => ({
+            const dashboardTabsMapping = dashboardTabs.map((c) => ({
                 uuid: c.uuid,
-                newUuid: newDashboardTabs[i].uuid,
+                newUuid: newDashboardTabs.find(
+                    (t) => t.dashboard_version_id === c.dashboard_version_id,
+                )?.uuid!,
```

Comment:
All tabs have the same `dashboard_version_id` so this returns the same uuid on each loop.
We can have something like you did for the other file, generate the new uuids before the insert loop.

Reply:
Hi @ZeRego you are right this logic is broken, i was too rush on the fix :(

Intention of this line is to handle a special case that, we have some tabbed dashboards copied when this fix is not in place, so in our `dashboard_tab` database table is bit messed up, leads to a result that the `tab_uuid` is not unique. but in this line for each tab row we generate new `tab_uuid` for it, so the old `tab_uuid` is mapped to many new `tab_uuid`, in this case only when you use new `tab_uuid` in combination with `dashboard_version_id` can make sure you can map back to the old original tab, i'v updated the code to flip the mapping logic to incorporate this case, let me know if that make sense to you.